### PR TITLE
docs: Add E2E testing tutorial.

### PR DESCRIPTION
**Pull request by victorgcramos**
This PR adds the documentation guide to write E2E tests, as well as the tutorial
to create mock api middlewares in order to provide a test environment without
needing to be connected to the backend.

### CODE REVIEW

File: docs/e2e.md [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+    cy.scrollTo("bottom");
+    cy.wait(1000);
+    // make sure no new proposals were fetched, since the inventory has only
+    // one token
+    cy.findAllByTestId("record-title").should("have.length", 1);
+  });
+});
+```
+
+Now, if we turned off the backend, the tests would still be able to render the
```
Comment (tiagoalvesdulce) [inline docs/e2e.md:237]: I think you could use the present tense here.

**Review by tiagoalvesdulce — APPROVED** [review]
This is really nice! Thanks for writing :)

File: docs/e2e.md [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+the expected user behavior. As you can see, we are making 2
+[assertions](https://docs.cypress.io/guides/core-concepts/introduction-to-cypress#Assertions)
+on our test case using the
+[`should`](https://docs.cypress.io/api/commands/should) command, which accepts
+[Chai chainers](https://docs.cypress.io/guides/references/assertions#Chai) as an
+argument.
+
+> Check the Cypress and Jest docs for more info.
+
+However, the most applications are not static, and you can't test it only
```
Comment (amass01) [inline docs/e2e.md:85]:
```suggestion
However, most of applications are not static, and you can't test them only
```

**Review by lukebp — APPROVED** [review]
This is great.

**Pull request merged**